mountpoint-s3/src/main.rs

```diff
@@ -12,92 +10,147 @@ use fuser::{MountOption, Session};
 use mountpoint_s3::fs::S3FilesystemConfig;
 use mountpoint_s3::fuse::session::FuseSession;
 use mountpoint_s3::fuse::S3FuseFilesystem;
-use mountpoint_s3::metrics::{metrics_tracing_span_layer, MetricsSink};
+use mountpoint_s3::metrics::MetricsSink;
 use mountpoint_s3::prefix::Prefix;
 use mountpoint_s3_client::{
     AddressingStyle, Endpoint, HeadBucketError, ObjectClientError, S3ClientConfig, S3CrtClient,
 };
 use mountpoint_s3_client::{ImdsCrtClient, S3ClientAuthConfig};
-use mountpoint_s3_crt::common::rust_log_adapter::RustLogAdapter;
 use nix::sys::signal::Signal;
 use nix::unistd::ForkResult;
 use regex::Regex;
-use time::format_description::FormatItem;
-use time::macros;
-use time::OffsetDateTime;
-use tracing_subscriber::{
-    filter::EnvFilter, filter::LevelFilter, layer::SubscriberExt, util::SubscriberInitExt, Layer,
-};
 
 mod build_info;
 
-fn init_tracing_subscriber(is_foreground: bool, log_directory: Option<&Path>) -> anyhow::Result<()> {
-    const LOG_DIRECTORY: &str = ".mountpoint-s3";
-    const LOG_FILE_NAME_FORMAT: &[FormatItem<'static>] =
-        macros::format_description!("mountpoint_s3_[year][month][day][hour][minute][second].log");
+mod logging {
```

Comment:
It's just organizational. `main.rs` is a little special in that it's not part of the crate (notice how all the imports mention `mountpoint_s3::` instead of just `crate::`), so giving it submodules works a little differently. We could probably move it to a file but I don't think it's worth it.